service/src/main/java/com/codeforcommunity/processor/PostsProcessor.java

```diff
@@ -76,6 +83,58 @@ public CommentsResponse getCommentsForPost(int postId) {
     return new CommentsResponse(comments);
   }
 
+  @Override
+  public void createPost(CreatePostRequest post) {
+    postTable.savePost(PostMapper.createRequestToRecord(post));
+  }
+
+  @Override
+  public void createComment(int postId, CreateCommentRequest comment) {
+    this.checkPostExists(postId);
+
+    commentTable.saveComment(CommentMapper.createRequestToRecord(postId, comment));
+  }
+
+  @Override
+  public void clapPost(int postId) {
+    this.checkPostExists(postId);
+
+    postTable.clapPost(postId);
+  }
+
+  @Override
+  public void clapComment(int postId, int commentId) {
+    this.checkPostExists(postId);
+
+    if (!commentTable.commentExists(postId, commentId)) {
```

Comment:
Why not just create an equivilant 'this.checkCommentExists' method?

Reply:
I don't have that in there because I only check if a comment exists twice, while I check if a post exists for almost every method. Do you think I should make one anyway to stay consistent?

Reply:
I think staying consistent is better. Also technically using it twice is enough for it to be worth pulling out anyways

Reply:
It has been added.